File: frontend/src/types/sentiment.ts
export interface SentimentResult {
  sentiment: string;
  confidence: number;
  is_relevant: boolean;
  relevance_score: number;
  relevance_explanation: string;
  matched_terms: Record<string, string[]>;
  feedback?: string;
}

export interface SentimentError {
  message: string;
  feedback?: string;
} 

File: backend/app/schemas/sentiment.py
from pydantic import BaseModel
from typing import Dict, Optional, List

class SentimentRequest(BaseModel):
    text: str

class SentimentResponse(BaseModel):
    sentiment: str
    confidence: float
    is_relevant: bool
    relevance_score: float
    relevance_explanation: str
    matched_terms: Dict[str, List[str]]
    feedback: Optional[str] = None 